scripts/bert/finetune_squad.py

```diff
@@ -404,23 +411,30 @@ def evaluate():
             max_seq_length=max_seq_length,
             doc_stride=doc_stride,
             max_query_length=max_query_length,
-            is_pad=True,
+            is_pad=False,
             is_training=False)._transform)
 
-    dev_data_transform = preprocess_dataset(dev_data, SQuADTransform(
-        berttoken,
-        max_seq_length=max_seq_length,
-        doc_stride=doc_stride,
-        max_query_length=max_query_length,
-        is_pad=True,
-        is_training=False))
+    dev_data_transform, data_dev_len = preprocess_dataset(
+        dev_data, SQuADTransform(
+            berttoken,
+            max_seq_length=max_seq_length,
+            doc_stride=doc_stride,
+            max_query_length=max_query_length,
+            is_pad=False,
+            is_training=False))
     log.info('The number of examples after preprocessing:{}'.format(
         len(dev_data_transform)))
 
+    batch_sampler = nlp.data.sampler.FixedBucketSampler(data_dev_len,
+                                                        batch_size=test_batch_size,
+                                                        num_buckets=10,
+                                                        ratio=0,
+                                                        shuffle=True)
+
```

Comment:
Because the validation data is generally small, the batch sampler is usually not used for validation. Even if needed, it is better to set ` shuffle ` to ` False `.

Reply:
Good point. For val data please set shuffle to False